### DISCUSSION TITLE
Message from DHARMa

### DISCUSSION THREAD

**Discussion by mrkozlov** [community/community#109604] [Programming Help]
### Select Topic Area

Product Feedback

### Body

residuals <- simulateResiduals(fittedModel = m_glM2)
Предупреждение:
В securityAssertion("nKcase - wrong dimensions of response") :
  Message from DHARMa: During the execution of a DHARMa function, some unexpected conditions occurred. Even if you didn't get an error, your results may not be reliable. Please check with the help if you use the functions as intended. If you think that the error is not on your side, I would be grateful if you could report the problem at https://github.com/florianhartig/DHARMa/issues 

 Context: nKcase - wrong dimensions of response
![Rplot](https://github.com/community/community/assets/102718905/4cd00c0f-94a0-47a1-b9c8-65cc1520d1cc)

**Comment by Ecosantos** [comment]
I faced the same problem recently. Then, I just realized what is going on:

I initially had a counting variable, however, because the range of these counts varies among the multiple variables I'm interested in, I z-scored to make them comparable among the numerous models I'm fitting. However, this new variable is no longer poison but does not resample Gaussian either, given the discrete repetitive values. I've included a small sample of this data below. 

### Original data

> dados[c(14:19)]%>%head()
  L2_L1 L3_L2 L4_L3 PUPA_L4 ADULTO_PUPA ADULTO_IMERSION
1     3     2     4       1           2              12
2     5     2     3       7           5              22
3     5     2     3       6           5              21
4     5     2     3       8           3              21
5     5     3     3       8           5              24
6     3     3     3       8           4              21



### Transformed data
> dados_std[c(14:19)]%>%head()
     L2_L1     L3_L2    L4_L3  PUPA_L4 ADULTO_PUPA ADULTO_IMERSION
1 1.098612 0.6931472 1.386294 0.000000   0.6931472        2.484907
2 1.609438 0.6931472 1.098612 1.945910   1.6094379        3.091042
3 1.609438 0.6931472 1.098612 1.791759   1.6094379        3.044522
4 1.609438 0.6931472 1.098612 2.079442   1.0986123        3.044522
5 1.609438 1.0986123 1.098612 2.079442   1.6094379        3.178054
6 1.098612 1.0986123 1.098612 2.079442   1.3862944        3.044522

**Reply by Reese-Chen** [reply]
So, what is the solution?

**Reply by Sanidhyafeaturist** [reply]
I encountered a warning while using the simulateResiduals function:

В securityAssertion("nKcase - wrong dimensions of response") :
Message from DHARMa: During the execution of a DHARMa function, some unexpected conditions occurred. Even if you didn't get an error, your results may not be reliable. Please check with the help if you use the functions as intended.
This suggests that there may be an issue with the response variable's dimensions in my fitted model (m_glM2). I will review the model specifications to ensure everything is set up correctly. If the issue persists, I’ll consider reporting it on the DHARMa GitHub page as suggested, providing details about my dataset and the steps leading to this warning.

**Reply by Ecosantos** [reply]
> So, what is the solution?

In my case I think this is a false red flag. The dataset is just small but is now gaussian indeed at least theoretically.

**Comment by mrkozlov** [comment]
I have forgot

сб, 5 окт. 2024 г., 12:12 Reese-Chen ***@***.***>:
